https://issues.redhat.com/browse/ACM-6260 Mirroring known issue (#5310)
open-cluster-management/rhacm-docs: clusters/release_notes/known_issues.adoc

//2.7:ACM-2664

When you use the console to add a new node pool to an existing hosted cluster, the same version of {ocp-short} might appear more than once in the list of options. You can select any instance in the list for the version that you want. 

[#mirror-ironic-agent-image]
=== Hub cluster {ocp} image is not mirrored automatically

> When you deploy a hub cluster on bare metal by using the Assisted Service, the hub cluster requires an ironic agent image. The hub cluster {ocp-short} version determines which ironic agent image is used.

Reviewer: ```suggestion
When you deploy a hub cluster on bare metal by using the Assisted Service, spoke cluster installation requires an ironic agent image. The hub cluster {ocp-short} version determines which ironic agent image is used.
```

Author: Thanks for the suggestion, I'll add installation but we cannot use "spoke cluster" in the docs, so I'll leave that out.

Reviewer: Sure, whatever term you need is fine. I just thought using `hub` made it sound like this was a requirement for installing the Assisted Service, when it's not. It's only required when you go to install an additional cluster.

Author: That's good to know, I'll make sure to clarify that, thanks!

//2.7:ACM-2664

When you use the console to add a new node pool to an existing hosted cluster, the same version of {ocp-short} might appear more than once in the list of options. You can select any instance in the list for the version that you want. 

[#mirror-ironic-agent-image]
=== Hub cluster {ocp} image is not mirrored automatically
//2.8:ACM-6260

> When you deploy a hub cluster on bare metal by using the Assisted Service, the hub cluster installation requires an ironic agent image. The hub cluster {ocp-short} version determines which ironic agent image is used.

Reviewer: This is still very confusing to me. Maybe something like this?

```suggestion
When you deploy a hub cluster on bare metal by using the Assisted Service, cluster installation requires an ironic agent image. The hub cluster {ocp-short} version determines which ironic agent image is used.
```

Author: Are we talking about managed clusters? I assumed it was hub cluster, but sounds like it should be managed cluster instead?

Reviewer: Yes, when installing a managed cluster using assisted we require this image.
It needs to be able to be pulled to the host that will become a node in a managed cluster.

I think spoke == managed

Author: That makes sense, I'll switch it to managed cluster! Thanks for clearing that up.

When you use the console to add a new node pool to an existing hosted cluster, the same version of {ocp-short} might appear more than once in the list of options. You can select any instance in the list for the version that you want. 

[#mirror-ironic-agent-image]
=== Hub cluster {ocp} image is not mirrored automatically
//2.8:ACM-6260

> If you deploy a hub cluster on bare metal by using the Assisted Service, installing managed clusters requires an ironic agent image. The hub cluster {ocp-short} version determines which ironic agent image is used.

Reviewer 2: I recommend restating the issue after you explain the cause/case. In this way you are letting the user know that this only effects users in disconnected environments earlier. I was a little confused reading, "The hub cluster {ocp-short} version determines which ironic agent image is used."

```suggestion
If you deploy a hub cluster on bare metal by using the Assisted Service, the {ocp} image is not mirrored automatically in disconnected environments. A matching ironic agent image of {ocp-short} must be used on your hub cluster to install managed clusters. The hub cluster {ocp-short} version detects which ironic agent image is being used.
```

After rereading your changes for a few times, it seems like the user does not have to create the image but they have to check if they have the correct mirrored image. What should the user do after they run the command and find that the agent image version doesn't match OCP ? I found this information in the credential_on_prem file:

"* For a disconnected environment, you must have a configured mirror registry where you can copy the release images for your cluster creation. See link:https://docs.openshift.com/container-platform/4.11/installing/disconnected_install/installing-mirroring-disconnected.html[Mirroring images for a disconnected installation] in the {ocp-short} documentation for more information."

If the user verifies that the images do not match, I would point them to a topic that helps that obtain a mirrored image. Heads up the OCP link is for 4.11, we would also need to update the reference link

Author: Thanks for the suggestions! Updated the PR, let me know if it looks good now.

When you use the console to add a new node pool to an existing hosted cluster, the same version of {ocp-short} might appear more than once in the list of options. You can select any instance in the list for the version that you want. 

[#mirror-ironic-agent-image]
=== Hub cluster {ocp} image is not mirrored automatically
//2.8:ACM-6260

If you deploy a hub cluster on bare metal by using the Assisted Service, the {ocp} image is not mirrored automatically in disconnected environments. A matching ironic agent image of {ocp-short} must be used on your hub cluster to install managed clusters. The hub cluster {ocp-short} version detects which ironic agent image is being used.

In a disconnected environment, you must mirror the {ocp-short} image manually. Make sure that the ironic agent image version matches the hub cluster {ocp-short} version by running the following command:

----
oc adm release info --image-for=ironic-agent <hub-release-image>
----

> If the image versions do not match, see link:https://docs.openshift.com/container-platform/4.12/installing/disconnected_install/installing-mirroring-disconnected.html[Mirroring images for a disconnected installation] in the {ocp-short} documentation to learn how to obtain a mirrored image. 

Reviewer 2: What are your thoughts about us adding a link to [Maintaining a custom list of release images while disconnected](https://github.com/stolostron/rhacm-docs/blob/2.8_stage/clusters/cluster_lifecycle/release_image_disconn.adoc). This way we promote our doc haha 

this is the link from the customer portal: https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html-single/clusters/index#release-images-disconnected

Author: I'm not familiar enough with this topic to say for sure. From the issue discussion it sounds like oc-mirror or oc adm is required but maybe someone from installer can chime in?

Update: I asked in the issue for confirmation.

Reviewer 2: @oafischer Yea, either install or cluster team

Reviewer: If we're trying to instruct someone to mirror just the single image I think something more simple like this would be the best option:

```
skopeo copy <image-from-oc-adm-output> <customer-mirror>
```

Reviewer: Alternatively if we'd rather them mirror the entire hub release image then we can probably link to some existing doc.

Author: Simple is better, thanks for the command. I'll use that instead of linking to the OCP doc.

[#mirror-ironic-agent-image]
=== Hub cluster {ocp} image is not mirrored automatically
//2.8:ACM-6260

If you deploy a hub cluster on bare metal by using the Assisted Service, the {ocp} image is not mirrored automatically in disconnected environments. A matching ironic agent image of {ocp-short} must be used on your hub cluster to install managed clusters. The hub cluster {ocp-short} version detects which ironic agent image is being used.

In a disconnected environment, you must mirror the {ocp-short} image manually. Make sure that the ironic agent image version matches the hub cluster {ocp-short} version by running the following command. Replace `<hub-release-image>` with the image for the hub cluster release:

----
oc adm release info --image-for=ironic-agent <hub-release-image>
----

If the image versions do not match, run the following command to mirror the {ocp-short} image manually:

----
skopeo copy <image-from-oc-adm-output> <mirror>
----

Replace `<image-from-oc-adm-output>` with the image from the output of `oc adm`.

> Replace `<mirror>` with your mirror.

Reviewer 2: ```suggestion
Replace `<mirror>` with your mirrored image.
```
